src/cargo/ops/cargo_add/dependency.rs

```diff
@@ -61,8 +62,13 @@ impl Dependency {
         }
     }
 
-    /// Set dependency to a given version
+    /// Set dependency to a given source and clear any conflicting fields
     pub fn set_source(mut self, source: impl Into<Source>) -> Self {
+        let source = source.into();
+        if let Source::Workspace(_) = source {
+            self.default_features = None;
+            self.rename = None;
```

Comment:
I debated heavily between clearing the fields or using `bail!`. I decided against clearing the fields as if you want to set a new source the idea is that you do not care about those fields. I can see how a `CargoResult<Self>` could be better and that should be an easy change to make if you feel it is better

Reply:
This left out another options: move `rename` into `name` so that the dependency table is as if the `package` field was dropped.  I was wondering about that vs what is here